File: library/src/main/java/io/ashdavies/rxtasks/CompletableEmitterSuccessListener.java
package io.ashdavies.rxtasks;

import com.google.android.gms.tasks.OnSuccessListener;
import io.reactivex.CompletableEmitter;

class CompletableEmitterSuccessListener implements OnSuccessListener<Void> {

  private final CompletableEmitter emitter;

  CompletableEmitterSuccessListener(CompletableEmitter emitter) {
    this.emitter = emitter;
  }

  @Override
  public void onSuccess(Void aVoid) {
    emitter.onComplete();
  }
}

File: src/main/kotlin/io/ashdavies/rx/rxtasks/CompletableEmitterSuccessListener.kt
package io.ashdavies.rx.rxtasks

import com.google.android.gms.tasks.OnSuccessListener
import io.reactivex.CompletableEmitter

internal class CompletableEmitterSuccessListener(private val emitter: CompletableEmitter) : OnSuccessListener<Void> {

  override fun onSuccess(void: Void) = emitter.onComplete()
}
